### PR TITLE
Visual end-to-end example and python3 compatibility

### PR DESCRIPTION
**Pull request by LemonPi**
This pull request adds a step by step guide on an artificial data set that's easy to visualize. An end-to-end training script is in `easy_example.py`.

It also brings python3 compatibility, centralizing where problem specific network architecture is defined, and bug fixes.

Addresses #14 and #13

### CODE REVIEW

**Comment by LemonPi** [comment]
Hey @shahsohil could you take a look at the changed files here? The example I added is in pytorch/easy_example. Other than that I refactored the code a bit and added python3 compatibility.

File: pytorch/DCCComputation.py [hunk 1]
```diff
@@ -54,7 +58,12 @@ def computeHyperParams(pairs, Z):
     numpairs = len(pairs)
     numsamples = len(Z)
     epsilon = np.linalg.norm(Z[pairs[:, 0].astype(int)] - Z[pairs[:, 1].astype(int)], axis=1)
-    epsilon = np.sort(epsilon[np.where(epsilon / np.sqrt(cfg.DIM) > cfg.RCC.NOISE_THRESHOLD)])
+    epsilon = np.sort(epsilon / np.sqrt(cfg.DIM))
+    # largest is noise just consider as noise
+    if epsilon[-1] < cfg.RCC.NOISE_THRESHOLD:
+        epsilon = np.asarray([cfg.RCC.NOISE_THRESHOLD])
+    else:
+        epsilon = epsilon[np.where(epsilon > cfg.RCC.NOISE_THRESHOLD)]
```
Comment (shahsohil) [inline pytorch/DCCComputation.py]: In the original code the 'epsilon' used was never factored by 1/np.sqrt(cfg.DIM). This factor was only used for selection. Can you please make this changes ?

File: pytorch/copyGraph.py [hunk 2]
```diff
@@ -3,25 +3,32 @@
...
 parser.add_argument('--data', dest='db', type=str, default='mnist', help='name of the dataset')
 parser.add_argument('--graph', dest='g', help='path to the graph file', default=None, type=str)
 parser.add_argument('--features', dest='feat', help='path to the feature file', default=None, type=str)
 parser.add_argument('--out', dest='out', help='path to the output file', default=None, type=str)
 parser.add_argument('--h5', dest='h5', action='store_true', help='to store as h5py file')
 
 if __name__ == '__main__':
-
     args = parser.parse_args()
+
```
Comment (shahsohil) [inline pytorch/copyGraph.py]: Should include main(args) ?

**Review by shahsohil — CHANGES_REQUESTED** [review]
I have left couple of comments. Everything else looks fine. Once it is addressed, will merge it.

File: pytorch/copyGraph.py (hunk 2 above)
Comment (LemonPi) [reply pytorch/copyGraph.py]: Done

File: pytorch/DCCComputation.py (hunk 1 above)
Comment (LemonPi) [reply pytorch/DCCComputation.py]: Ok changed back

**Comment by LemonPi** [comment]
@shahsohil made those requested changes

**Review by shahsohil — APPROVED** [review]
Thanks @LemonPi
